File: plugins/NMaven.Plugin.Devenv/src/main/csharp/NMaven/Plugin/Devenv/DevenvMojo.cs
//
// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.
//

using System;
using System.Collections.Generic;
using System.IO;
using System.Diagnostics;

using Microsoft.Win32;

using NMaven.Plugin;

namespace NMaven.Plugin.Devenv
{
	/// <summary>
	/// Description
	/// </summary>
	[Serializable]
	[ClassAttribute(Phase = "deploy", Goal = "start")]
	public sealed class DevenvMojo : AbstractMojo
	{
		public DevenvMojo()
		{
		}
		
		[FieldAttribute("artifactId", Expression = "${project.artifactId}", Type = "java.lang.String")]
		public String artifactId;

        [FieldAttribute("buildDirectory", Expression = "${project.build.directory}", Type = "java.lang.String")]
        public String buildDirectory;
		
		public override Type GetMojoImplementationType()
		{
			return this.GetType();
		}
		
		public override void Execute()
		{
            string args = "/ResetAddin " + artifactId + ".Connect " + "/Log " + @"""" + @buildDirectory 
                + @"\VisualStudio.log" + @"""";
            RegistryKey visualStudioKey = 
                Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\VisualStudio\8.0");
            String installDir = (String) visualStudioKey.GetValue("InstallDir");
            ProcessStartInfo processStartInfo =
                new ProcessStartInfo(@installDir + "devenv.exe", args);
            Process.Start(processStartInfo);
		}
	}
}

File: plugins/NMaven.Plugin.Devenv/src/main/java/NMaven/Plugin/Devenv/DevenvMojo.java
package NMaven.Plugin.Devenv;

import org.apache.maven.dotnet.plugin.FieldAnnotation;

/**
 * @phase deploy
 * @goal start
 */
public class DevenvMojo
    extends org.apache.maven.dotnet.plugin.AbstractMojo
{
       /**
        * @parameter expression = "${project.artifactId}"
        */
        @FieldAnnotation()
        public java.lang.String artifactId;

       /**
        * @parameter expression = "${project.build.directory}"
        */
        @FieldAnnotation()
        public java.lang.String buildDirectory;

       /**
        * @parameter expression = "${project}"
        */
        private org.apache.maven.project.MavenProject project;

       /**
        * @parameter expression = "${settings.localRepository}"
        */
        private String localRepository;

       /**
        * @parameter expression = "${vendor}"
        */
        private String vendor;

       /**
        * @parameter expression = "${vendorVersion}"
        */
        private String vendorVersion;

       /**
        * @parameter expression = "${frameworkVersion}"
        */
        private String frameworkVersion;

       /**
        * @component
        */
        private org.apache.maven.dotnet.executable.NetExecutableFactory netExecutableFactory;

       /**
        * @component
        */
        private org.apache.maven.dotnet.plugin.PluginContext pluginContext;

        public String getMojoArtifactId()
        {
            return "NMaven.Plugin.Devenv";
        }

        public String getMojoGroupId()
        {
            return "NMaven.Plugins";
        }

        public String getClassName()
        {
            return "NMaven.Plugin.Devenv.DevenvMojo";
        }

        public org.apache.maven.dotnet.plugin.PluginContext getNetPluginContext()
        {
            return pluginContext;
        }

        public org.apache.maven.dotnet.executable.NetExecutableFactory getNetExecutableFactory()
        {
            return netExecutableFactory;
        }

        public org.apache.maven.project.MavenProject getMavenProject()
        {
            return project;
        }

        public String getLocalRepository()
        {
            return localRepository;
        }

        public String getVendorVersion()
        {
            return vendorVersion;
        }

        public String getVendor()
        {
            return vendor;
        }

        public String getFrameworkVersion()
        {
            return frameworkVersion;
        }

}
